api/r0/upload.go

```diff
@@ -22,14 +25,52 @@ type MediaUploadedResponse struct {
 	Blurhash   string `json:"xyz.amorgan.blurhash,omitempty"`
 }
 
+type MediaCreatedResponse struct {
+	ContentUri      string `json:"content_uri"`
+	UnusedExpiresAt int64  `json:"unused_expires_at"`
+}
+
+func CreateMedia(r *http.Request, rctx rcontext.RequestContext, user api.UserInfo) interface{} {
```

Comment:
This isn't r0, so should be in the `unstable` directory instead (alongside the media info and local copy endpoints).